File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/productsection/activities/ProductList.java
package com.baytrendy.shopifyapp.productsection.activities;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000l\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u000b\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010!\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0005\u0018\u00002\u00020\u0001:\u0001,B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0010\u0010\"\u001a\u00020#2\u0006\u0010$\u001a\u00020%H\u0002J\u0012\u0010&\u001a\u00020#2\b\u0010\'\u001a\u0004\u0018\u00010(H\u0014J\u0016\u0010)\u001a\u00020#2\f\u0010\u0019\u001a\b\u0012\u0004\u0012\u00020\u001b0\u001aH\u0002J\u0010\u0010*\u001a\u00020#2\u0006\u0010+\u001a\u00020\u0016H\u0002R\u001e\u0010\u0003\u001a\u00020\u00048\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0005\u0010\u0006\"\u0004\b\u0007\u0010\bR\u0010\u0010\t\u001a\u0004\u0018\u00010\nX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001e\u0010\u000b\u001a\u00020\f8\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\r\u0010\u000e\"\u0004\b\u000f\u0010\u0010R\u000e\u0010\u0011\u001a\u00020\u0012X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0013\u001a\u0004\u0018\u00010\u0014X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0015\u001a\u0004\u0018\u00010\u0016X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0017\u001a\u0004\u0018\u00010\u0018X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0016\u0010\u0019\u001a\n\u0012\u0004\u0012\u00020\u001b\u0018\u00010\u001aX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u000e\u0010\u001c\u001a\u00020\u001dX\u0082\u0004\u00a2\u0006\u0002\n\u0000R\u0014\u0010\u001e\u001a\b\u0012\u0002\b\u0003\u0018\u00010\u001fX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010 \u001a\u0004\u0018\u00010!X\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006-"}, d2 = {"Lcom/shopify/shopifyapp/productsection/activities/ProductList;", "Lcom/shopify/shopifyapp/basesection/activities/BaseActivity;", "()V", "adapter", "Lcom/shopify/shopifyapp/productsection/adapters/ProductRecylerAdapter;", "getAdapter", "()Lcom/shopify/shopifyapp/productsection/adapters/ProductRecylerAdapter;", "setAdapter", "(Lcom/shopify/shopifyapp/productsection/adapters/ProductRecylerAdapter;)V", "binding", "Lcom/shopify/shopifyapp/base/databinding/MProductlistitemBinding;", "factory", "Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "getFactory", "()Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "setFactory", "(Lcom/shopify/shopifyapp/utils/ViewModelFactory;)V", "flag", "", "productListModel", "Lcom/shopify/shopifyapp/productsection/viewmodels/ProductListModel;", "productcursor", "", "productlist", "Landroidx/recyclerview/widget/RecyclerView;", "products", "", "Lcom/shopify/buy3/Storefront$ProductEdge;", "recyclerViewOnScrollListener", "Landroidx/recyclerview/widget/RecyclerView$OnScrollListener;", "sheet", "Lcom/google/android/material/bottomsheet/BottomSheetBehavior;", "sortselction", "Landroid/widget/RadioGroup;", "appySort", "", "radioButton", "Lcom/shopify/shopifyapp/customviews/MageNativeRadioButton;", "onCreate", "savedInstanceState", "Landroid/os/Bundle;", "setRecylerData", "showToast", "msg", "Handler", "base_debug"})
public final class ProductList extends com.baytrendy.shopifyapp.basesection.activities.BaseActivity {
    private com.baytrendy.shopifyapp.databinding.MProductlistitemBinding binding;
    private androidx.recyclerview.widget.RecyclerView productlist;
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.utils.ViewModelFactory factory;
    private com.baytrendy.shopifyapp.productsection.viewmodels.ProductListModel productListModel;
    private java.util.List<com.shopify.buy3.Storefront.ProductEdge> products;
    private java.lang.String productcursor;
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.productsection.adapters.ProductRecylerAdapter adapter;
    private com.google.android.material.bottomsheet.BottomSheetBehavior<?> sheet;
    private android.widget.RadioGroup sortselction;
    private boolean flag;
    private final androidx.recyclerview.widget.RecyclerView.OnScrollListener recyclerViewOnScrollListener = null;
    private java.util.HashMap _$_findViewCache;
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.utils.ViewModelFactory getFactory() {
        return null;
    }
    
    public final void setFactory(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.utils.ViewModelFactory p0) {
    }
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.productsection.adapters.ProductRecylerAdapter getAdapter() {
        return null;
    }
    
    public final void setAdapter(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.productsection.adapters.ProductRecylerAdapter p0) {
    }
    
    @java.lang.Override()
    protected void onCreate(@org.jetbrains.annotations.Nullable()
    android.os.Bundle savedInstanceState) {
    }
    
    private final void showToast(java.lang.String msg) {
    }
    
    private final void appySort(com.baytrendy.shopifyapp.customviews.MageNativeRadioButton radioButton) {
    }
    
    private final void setRecylerData(java.util.List<com.shopify.buy3.Storefront.ProductEdge> products) {
    }
    
    public ProductList() {
        super();
    }
    
    @kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000\u0018\n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\b\u0086\u0004\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u000e\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u0006\u00a8\u0006\u0007"}, d2 = {"Lcom/shopify/shopifyapp/productsection/activities/ProductList$Handler;", "", "(Lcom/shopify/shopifyapp/productsection/activities/ProductList;)V", "openSort", "", "view", "Landroid/view/View;", "base_debug"})
    public final class Handler {
        
        public final void openSort(@org.jetbrains.annotations.NotNull()
        android.view.View view) {
        }
        
        public Handler() {
            super();
        }
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/productsection/activities/ProductList.kt
package com.baytrendy.shopifyapp.productsection.activities

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.View
import android.view.ViewGroup
import android.widget.RadioGroup
import android.widget.TextView
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.shopify.buy3.Storefront
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MProductlistitemBinding
import com.baytrendy.shopifyapp.basesection.activities.NewBaseActivity
import com.baytrendy.shopifyapp.cartsection.activities.CartList
import com.baytrendy.shopifyapp.customviews.MageNativeRadioButton
import com.baytrendy.shopifyapp.productsection.adapters.ProductRecylerAdapter
import com.baytrendy.shopifyapp.productsection.viewmodels.ProductListModel
import com.baytrendy.shopifyapp.utils.Constant
import com.baytrendy.shopifyapp.utils.ViewModelFactory

import javax.inject.Inject

class ProductList : NewBaseActivity() {
    private var binding: MProductlistitemBinding? = null
    private var productlist: RecyclerView? = null

    @Inject
    lateinit var factory: ViewModelFactory
    private var productListModel: ProductListModel? = null
    private var products: MutableList<Storefront.ProductEdge>? = null
    private var productcursor: String? = null

    @Inject
    lateinit var product_adapter: ProductRecylerAdapter
    private var sheet: BottomSheetBehavior<*>? = null
    private var sortselction: RadioGroup? = null
    private var flag = true
    private val recyclerViewOnScrollListener = object : RecyclerView.OnScrollListener() {
        override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
            super.onScrollStateChanged(recyclerView, newState)
        }

        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
            super.onScrolled(recyclerView, dx, dy)
            val visibleItemCount = recyclerView.layoutManager!!.childCount
            val totalItemCount = recyclerView.layoutManager!!.itemCount
            var firstVisibleItemPosition = 0
            if (recyclerView.layoutManager is LinearLayoutManager) {
                firstVisibleItemPosition = (recyclerView.layoutManager as LinearLayoutManager).findFirstVisibleItemPosition()
            } else if (recyclerView.layoutManager is GridLayoutManager) {
                firstVisibleItemPosition = (recyclerView.layoutManager as GridLayoutManager).findFirstVisibleItemPosition()
            }
            if (!recyclerView.canScrollVertically(1)) {
                if (visibleItemCount + firstVisibleItemPosition >= totalItemCount && firstVisibleItemPosition > 0
                        && totalItemCount >= products!!.size) {
                    productListModel!!.number = 20
                    productListModel!!.cursor = productcursor!!
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val group = findViewById<ViewGroup>(R.id.container)
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.m_productlistitem, group, true)
        binding!!.handler = Handler()
        productlist = setLayout(binding!!.root.findViewById(R.id.productlist), "grid")
        sheet = BottomSheetBehavior.from(binding!!.root.findViewById<View>(R.id.bottom_sheet))
        sortselction = binding!!.root.findViewById(R.id.sortselction)
        showBackButton()
        if (intent.hasExtra("tittle") && intent.getStringExtra("tittle") != null) {
            showTittle(intent.getStringExtra("tittle"))
        }
        //setPanelBackgroundColor(HomePageViewModel.panel_bg_color!!)
        (application as MyApplication).mageNativeAppComponent!!.doProductListInjection(this)
        productListModel = ViewModelProvider(this, factory).get(ProductListModel::class.java)
        productListModel!!.context = this
        productListModel?.collectionData?.observe(this, Observer { this.collectionResponse(it) })
        if (intent.getStringExtra("ID") != null) {
            productListModel!!.setcategoryID(intent.getStringExtra("ID"))
        }
        if (intent.getStringExtra("handle") != null) {
            productListModel!!.setcategoryHandle(intent.getStringExtra("handle"))
        }
        if (intent.getStringExtra("ID") == null && intent.getStringExtra("handle") == null) {
            productListModel!!.shopID = "allproduct"
            flag = false
        }
        productListModel!!.message.observe(this, Observer { this.showToast(it) })
        productListModel!!.Response()
        productListModel!!.filteredproducts.observe(this, Observer<MutableList<Storefront.ProductEdge>> { this.setRecylerData(it) })
        productlist!!.addOnScrollListener(recyclerViewOnScrollListener)
        sortselction!!.setOnCheckedChangeListener { group, checkedId ->
            val selectedId = group.checkedRadioButtonId
            val radioButton = findViewById<View>(selectedId) as MageNativeRadioButton
            appySort(radioButton)
        }

        binding!!.view.setOnClickListener() {
            if (sheet!!.state == BottomSheetBehavior.STATE_EXPANDED) {
                sheet!!.state = BottomSheetBehavior.STATE_COLLAPSED
            }
        }
        sheet!!.addBottomSheetCallback(object : BottomSheetBehavior.BottomSheetCallback() {
            override fun onStateChanged(bottomSheet: View, newState: Int) {
                if (newState == BottomSheetBehavior.STATE_DRAGGING) {
                    sheet!!.state = BottomSheetBehavior.STATE_EXPANDED
                }
                if (newState == BottomSheetBehavior.STATE_EXPANDED) {
                    binding!!.view.visibility = View.VISIBLE
                }
                if (newState == BottomSheetBehavior.STATE_COLLAPSED) {
                    binding!!.view.visibility = View.GONE
                }
            }

            override fun onSlide(bottomSheet: View, slideOffset: Float) {
                binding!!.view.visibility = View.VISIBLE
                binding!!.view.alpha = slideOffset
            }
        })
    }

    private fun collectionResponse(it: Storefront.Collection?) {
        if (it?.title != null) {
            showTittle(it?.title)
        }
    }

    override fun onResume() {
        super.onResume()
        if (textView != null) {
            textView!!.text = "" + productListModel!!.cartCount
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.m_product, menu)
        val item = menu.findItem(R.id.cart_item)
        item.setActionView(R.layout.m_count)
        val notifCount = item.actionView
        textView = notifCount.findViewById<TextView>(R.id.count)
        textView?.text = "" + cartCount
        notifCount.setOnClickListener {
            val mycartlist = Intent(this, CartList::class.java)
            startActivity(mycartlist)
            Constant.activityTransition(this)
        }
        return true
    }

    private fun showToast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_LONG).show()
    }

    private fun appySort(radioButton: MageNativeRadioButton) {
        try {

            when (radioButton.tag.toString()) {
                "atoz" -> {
                    if (flag) {
                        productListModel!!.sortKeys = Storefront.ProductCollectionSortKeys.TITLE
                    } else {
                        productListModel!!.keys = Storefront.ProductSortKeys.TITLE
                    }
                    productListModel!!.isDirection = false
                }
                "ztoa" -> {
                    if (flag) {
                        productListModel!!.sortKeys = Storefront.ProductCollectionSortKeys.TITLE
                    } else {
                        productListModel!!.keys = Storefront.ProductSortKeys.TITLE
                    }
                    productListModel!!.isDirection = true
                }
                "htol" -> {
                    if (flag) {
                        productListModel!!.sortKeys = Storefront.ProductCollectionSortKeys.PRICE
                    } else {
                        productListModel!!.keys = Storefront.ProductSortKeys.PRICE
                    }
                    productListModel!!.isDirection = true
                }
                "ltoh" -> {
                    if (flag) {
                        productListModel!!.sortKeys = Storefront.ProductCollectionSortKeys.PRICE
                    } else {
                        productListModel!!.keys = Storefront.ProductSortKeys.PRICE
                    }
                    productListModel!!.isDirection = false
                }
            }
            products = null
            productListModel!!.number = 10
            productListModel!!.cursor = "nocursor"
            sheet!!.state = BottomSheetBehavior.STATE_COLLAPSED
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    private fun setRecylerData(products: MutableList<Storefront.ProductEdge>) {
        try {
            if (products.size > 0) {
                product_adapter!!.presentmentcurrency = productListModel!!.presentmentCurrency
                if (this.products == null) {
                    this.products = products
                    product_adapter!!.setData(this.products, this@ProductList, productListModel!!.repository)
                    productlist!!.adapter = product_adapter
                } else {
                    this.products!!.addAll(products)
                    product_adapter!!.notifyDataSetChanged()
                }
                productcursor = this.products!![this.products!!.size - 1].cursor
                Log.i("MageNative", "Cursor : " + productcursor!!)
            } else {
                showToast(resources.getString(R.string.noproducts))
            }

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    inner class Handler {
        fun openSort(view: View) {
            if (sheet!!.state != BottomSheetBehavior.STATE_EXPANDED) {
                sheet!!.setState(BottomSheetBehavior.STATE_EXPANDED)
                sortselction?.visibility = View.VISIBLE
            } else {
                sheet!!.setState(BottomSheetBehavior.STATE_COLLAPSED)
                sortselction?.visibility = View.GONE
            }
        }
    }
}
